base/synchronization/waitable_event_mac.cc

```diff
@@ -39,8 +39,10 @@ WaitableEvent::WaitableEvent(ResetPolicy reset_policy,
                              InitialState initial_state)
     : policy_(reset_policy) {
   // Pointer registration is needed for sorting in WaitSet.user_events_
-  recordreplay::RegisterPointer("WaitableEvent", this);
-  record_replay_ordered_lock_id_ = recordreplay::CreateOrderedLock("WaitableEvent");
+  if (!recordreplay::AreEventsDisallowed() || recordreplay::HasDivergedFromRecording()) {
```

Comment:
I'm amazed by how little changes are needed for mac. i.e. it looks like even this `cc` file is mac specific

Reply:
Well, there is some other mac specific logic already in place, since we used to be able to record/replay on chrome 91 and I made sure that the mac build worked during the 108 rebase.